Fails to build with `-fno-common` / GCC 10

<!--
When creating a bug report please:
- Verify first that your issue is not already reported on GitHub.
- Test if the latest release and master branch are affected too.
-->

##### Bug report summary
gcc-10 and above flipped a default from -fcommon to -fno-common: https://gcc.gnu.org/PR85678

netdata currently fails to compile when given the CFLAG `-fno-common`:
```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/libnetdata.o (symbol from plugin): in function `strdupz':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/locks/locks.o (symbol from plugin): in function `netdata_thread_disable_cancelability':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/log/log.o (symbol from plugin): in function `facility_log':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/popen/popen.o (symbol from plugin): in function `myp_init':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/procfile/procfile.o (symbol from plugin): in function `procfile_adaptive_initial_allocation':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/os.o (symbol from plugin): in function `get_system_cpus':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/simple_pattern/simple_pattern.o (symbol from plugin): in function `simple_pattern_create':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/socket/socket.o (symbol from plugin): in function `poll_default_add_callback':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/socket/security.o (symbol from plugin): in function `security_openssl_library':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/statistical/statistical.o (symbol from plugin): in function `log_series_to_stderr':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/storage_number/storage_number.o (symbol from plugin): in function `pack_storage_number':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/threads/threads.o (symbol from plugin): in function `netdata_thread_tag':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/url/url.o (symbol from plugin): in function `from_hex':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/json/json.o (symbol from plugin): in function `json_callback_print':
(.text+0x0): multiple definition of `silencers_filename'; libnetdata/log/log.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/json/json.o (symbol from plugin): in function `json_callback_print':
(.text+0x0): multiple definition of `netdata_ready'; libnetdata/log/log.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/json/json.o (symbol from plugin): in function `json_callback_print':
(.text+0x0): multiple definition of `api_secret'; libnetdata/log/log.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/json/json.o (symbol from plugin): in function `json_callback_print':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: libnetdata/health/health.o (symbol from plugin): in function `create_silencer':
(.text+0x0): multiple definition of `silencers'; collectors/apps.plugin/apps_plugin.o (symbol from plugin):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
```

Here's the the build log showing the failure:
[net-analyzer:netdata-9999:20200127-204109.log](https://github.com/netdata/netdata/files/4119330/net-analyzer.netdata-9999.20200127-204109.log)


Here's some more info about how to fix the issue: https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common
##### OS / Environment
<!--
Provide as much information about your environment (which operating system and distribution you're using, if Netdata is running in a container, etc.)
as possible to allow us reproduce this bug faster.

To get this information, execute the following commands based on your operating system:
- uname -a; grep -Hv "^#" /etc/*release  # Linux
- uname -a; uname -K                     # BSD
- uname -a; sw_vers                      # macOS

Place the output from the command in the code section below.  
 -->
n/a, use GCC 10 or specify the `fno-common` CFLAG.

##### Netdata version
<!--
Provide output of `netdata -V`.
 
If Netdata is running, execute: $(ps aux | grep -E -o "[a-zA-Z/]+netdata ") -V
 -->
 Current master, https://github.com/netdata/netdata/commit/a4223457b49497f57bd199dfaf5cf35c4fb86997

##### Component Name
<!--
Let us know which component is affected by the bug. Our code is structured according to its component,
so the component name is the same as the top level directory of the repository.
For example, a bug in the dashboard would be under the web component.
-->
core

##### Steps To Reproduce
<!--
Describe how you found this bug and how we can reproduce it, preferably with a minimal test-case scenario.
If you'd like to attach larger files, use gist.github.com and paste in links.
-->

1. Compile with CFLAGS="-fno-common"

##### Expected behavior
<!-- Provide a clear and concise description of what you expected to happen. -->
Should compile.

--- a/daemon/common.c
+++ b/daemon/common.c
@@ -13,4 +13,4 @@ char *netdata_configured_varlib_dir          = VARLIB_DIR;
 char *netdata_configured_home_dir            = CACHE_DIR;
 char *netdata_configured_host_prefix         = NULL;
 char *netdata_configured_timezone            = NULL;
-
+int netdata_ready;

--- a/daemon/common.h
+++ b/daemon/common.h
@@ -84,6 +84,6 @@ extern char *netdata_configured_timezone;
 extern int netdata_zero_metrics_enabled;
 extern int netdata_anonymous_statistics_enabled;
 
-int netdata_ready;
+extern int netdata_ready;
 
 #endif /* NETDATA_COMMON_H */

--- a/health/health.c
+++ b/health/health.c
@@ -9,6 +9,7 @@ struct health_cmdapi_thread_status {
 };
 
 unsigned int default_health_enabled = 1;
+char *silencers_filename;
 
 // ----------------------------------------------------------------------------
 // health initialization

--- a/health/health.h
+++ b/health/health.h
@@ -52,7 +52,7 @@ extern unsigned int default_health_enabled;
 
 #define HEALTH_SILENCERS_MAX_FILE_LEN 10000
 
-char *silencers_filename;
+extern char *silencers_filename;
 
 extern void health_init(void);
 extern void *health_main(void *ptr);

--- a/libnetdata/health/health.c
+++ b/libnetdata/health/health.c
@@ -1,5 +1,7 @@
 #include "health.h"
 
+SILENCERS *silencers;
+
 /**
  * Create Silencer
  *

--- a/libnetdata/health/health.h
+++ b/libnetdata/health/health.h
@@ -44,7 +44,7 @@ typedef struct silencers {
     SILENCER *silencers;
 } SILENCERS;
 
-SILENCERS *silencers;
+extern SILENCERS *silencers;
 
 extern SILENCER *create_silencer(void);
 extern int health_silencers_json_read_callback(JSON_ENTRY *e);

--- a/web/api/web_api_v1.c
+++ b/web/api/web_api_v1.c
@@ -2,6 +2,8 @@
 
 #include "web_api_v1.h"
 
+char *api_secret;
+
 static struct {
     const char *name;
     uint32_t hash;

--- a/web/api/web_api_v1.h
+++ b/web/api/web_api_v1.h
@@ -29,6 +29,6 @@ extern void host_labels2json(RRDHOST *host, BUFFER *wb, size_t indentation);
 extern void web_client_api_v1_init(void);
 extern void web_client_api_v1_management_init(void);
 
-char *api_secret;
+extern char *api_secret;
 
 #endif //NETDATA_WEB_API_V1_H

